### PR TITLE
Introducing mkdocs generated github pages for etcd-druid documentation

### LINKED ISSUE #897: Documentation improvement - Part 2
**How to categorize this issue?**

/area documentation
/kind enhancement

**What would you like to be added**:
https://github.com/gardener/etcd-druid/issues/813 got closed prematurely. Most of the documentation changes have been done but some are still pending. This issue captures the pending items.

- [ ] Use [mkdocs](https://squidfunk.github.io/mkdocs-material) to generate github pages for documentation (https://github.com/gardener/etcd-druid/pull/909)
- [ ] Add `docs/usage/copy-backups.md`
- [ ] Add a documentation to explain auto-single-member restoration under `docs/concepts`
- [ ] Add doc to explain snapshot compaction under `docs/concepts`
- [ ] Add API reference docs and include them in mkdocs

**Why is this needed**:
To improve ease of consumption.

### PR DESCRIPTION
**Pull request by unmarshall**
**How to categorize this PR?**

/area documentation
/kind enhancement

**What this PR does / why we need it**:
Following is done in this PR:

1. Small corrections to docs
2. Introduced mkdocs specific configuration (YAML and CSS)
3. Added github action to publish the docs
4. Adapted docs to use mkdocs plugins

**Which issue(s) this PR fixes**:
Partially fixes #897 

**Special notes for your reviewer**:
Once this PR is reviewed and merged there are some additional steps that needs to be done:
- Create an orphan and empty `gh-pages` branch. To create such a branch do the following:
```
git checkout --orhpan gh-pages
git rm -rf .
git commit --allow-empty -m "<commit-msg>"
git push origin gh-pages
```
- Configure github pages to use `gh-pages` branch

To view the docs you can see this in the fork https://unmarshall.github.io/etcd-druid/index.html

**Release note**:

```other operator
Introduced github pages to host the etcd-druid documentation
```

### CODE REVIEW

**Review by ashwani2k — APPROVED** [review]
Looks good to me except that now the !!! Note instead of > Note is not the same when rendered in a markdown viewer. 
But as our goal is for the people to read through mkdocs, I'm guessing its fine to live with that.

File: docs/proposals/05-etcd-operator-tasks.md
Comment (seshachalam-yv) [inline docs/proposals/05-etcd-operator-tasks.md:16]: If you're removing the Table of Contents, please also remove it from the [proposal template](https://github.com/gardener/etcd-druid/blob/master/docs/proposals/00-template.md).

**Review by seshachalam-yv — CHANGES_REQUESTED** [review]

File: docs/proposals/05-etcd-operator-tasks.md
Comment (unmarshall) [reply docs/proposals/05-etcd-operator-tasks.md:16]: Done

File: docs/development/prepare-dev-environment.md
Comment (renormalize) [inline docs/development/prepare-dev-environment.md:9]: Could you also add a couple lines for Linux which suggests using any popular package manager to install such dependencies?

**Review by seshachalam-yv — APPROVED** [review]
lgtm

File: .github/workflows/publish-docs.yml
Comment (renormalize) [inline .github/workflows/publish-docs.yml:29]: Nit:
Does GitHub accept the file extension to be `.yaml` instead of `.yml`? If yes, can the file be renamed to `publish-docs.yaml`?
I ask since all yaml files in the repository use the `.yaml` extension.

File: docs/deployment/getting-started-locally/getting-started-locally.md
Comment (renormalize) [inline docs/deployment/getting-started-locally/getting-started-locally.md:36]: Can you remove the prepending numbers for each section in the entire file? It is not consistent with the rest of the documentation.

```suggestion
## Setting up etcd-druid
```

File: docs/usage/recovering-etcd-clusters.md
Comment (renormalize) [inline docs/usage/recovering-etcd-clusters.md:41]: Can you remove the prepending numbers for each section in the entire file? It is not consistent with the rest of the documentation.

```suggestion
#### Prepare Etcd Resource to allow manual updates
```

File: docs/development/raising-a-pr.md
Comment (renormalize) [inline docs/development/raising-a-pr.md:45]: Can you remove the prepending numbers for each section in the entire file? It is not consistent with the rest of the documentation.

**Comment by renormalize** [comment]
I'm posting my comments in 2, please do not merge.

**Review by renormalize — CHANGES_REQUESTED** [review]
Thanks for the improvements!

Can the same font (if not font, family) be used throughout? The font in the menu bar and title doesn't really look cohesive with the font used for the content.

Screenshot 2024-11-11 at 10 55 43

File: mkdocs.yml
Comment (renormalize) [inline mkdocs.yml:1]: https://github.com/gardener/etcd-druid/pull/909#discussion_r1835969619

Also, are the trailing newlines necessary?

**Review by renormalize — CHANGES_REQUESTED** [review]

**Comment by renormalize** [comment]
Please edit the commit message before merging to avoid "WIP" commit messages in the commit history of the repo. Thanks.

File: docs/deployment/getting-started-locally/manage-s3-emulator.md
Comment (unmarshall) [reply docs/deployment/getting-started-locally/manage-s3-emulator.md:56]: Ah this was corrected by Anvesh and i took those changes already. Will check again.

File: docs/deployment/getting-started-locally/getting-started-locally.md
Comment (unmarshall) [reply docs/deployment/getting-started-locally/getting-started-locally.md:36]: I prefer having numbers. This document provide a sequence of steps as compared to other documents.

File: docs/development/prepare-dev-environment.md
Comment (unmarshall) [reply docs/development/prepare-dev-environment.md:9]: The title says that the section is for macOS only.

File: docs/development/raising-a-pr.md
Comment (unmarshall) [reply docs/development/raising-a-pr.md:45]: Numbers are on purpose.

File: docs/usage/recovering-etcd-clusters.md
Comment (unmarshall) [reply docs/usage/recovering-etcd-clusters.md:41]: Numbers are on purpose

File: .github/workflows/publish-docs.yml
Comment (unmarshall) [reply .github/workflows/publish-docs.yml:29]: I have followed the naming convention from all yaml files within .github
See https://github.com/gardener/etcd-druid/blob/master/.github/dependabot.yml

File: mkdocs.yml
Comment (unmarshall) [reply mkdocs.yml:1]: removed

File: .github/workflows/publish-docs.yml
Comment (renormalize) [reply .github/workflows/publish-docs.yml:29]: Alright. Let's leave this as is for now.

**Review by renormalize — APPROVED** [review]

**Pull request merged**
